pallets/loans/src/types/mod.rs

```diff
@@ -85,25 +85,45 @@ pub enum MutationError {
 	DiscountedCashFlowExpected,
 	/// Emits when a modification expect the loan to have an iternal pricing.
 	InternalPricingExpected,
+	/// Maturity extensions exceed max extension allowed.
+	MaturityExtendedToMuch,
 }
 
 /// Specify the expected repayments date
 #[derive(Encode, Decode, Clone, PartialEq, Eq, TypeInfo, RuntimeDebug, MaxEncodedLen)]
 pub enum Maturity {
 	/// Fixed point in time, in secs
-	Fixed(Moment),
+	Fixed {
+		/// Secs when maturity ends
+		date: Moment,
+		/// Date extension without special permissions
+		extension: Moment,
```

Comment:
We definitely need better types to differentiate different time units. I think for loans, we would need a `Seconds` type with a bound of `From<Moment>`. Or use this trait instead: #1360. I think this is even more important than differentiating `Instant` from `Duration`, and more expressive for the UI to know what the unit is.